File: src/java/4/trunk/src/main/java/com/mycompany/Release4.java
package com.mycompany;

/**
 * @author Paul Hammant DevOps, (c) 2018
 */
public class Release4 extends ReleaseToggles {

    static int colorCtr = 0;

    enum Color {

        Blonde(),
        Brown(),
        Black(),
        Red();

        Color() {
        }

        static Color rotatingChoice() {
            return Color.values()[colorCtr++ % Color.values().length];
        }

    }

    @Override
    public Object getChangingHairColor() {
        return Color.rotatingChoice();
    }
}

File: src/dotnet/4/trunk/Release4.cs
namespace MyCompany;

/// <author>Paul Hammant DevOps, (c) 2018</author>
public class Release4 : IReleaseToggles
{
    public enum Color
    {
        Blonde,
        Brown,
        Black,
        Red
    }

    private static int _colorCtr = 0;

    public static Color RotatingChoice()
    {
        var values = Enum.GetValues<Color>();
        return values[_colorCtr++ % values.Length];
    }

    public object ChangingHairColor() => RotatingChoice();
}
